contribute/design/1.x/maya/20190107-cStor-volume-resize.md

```diff
@@ -0,0 +1,259 @@
+    and also check capacity changes in CV and CVC object if so update capacity
+    field and status field with required details like (size, pending,
+    LastTransitionTime, LastProbeTime) in CV object.
+  - During reconciliation time Check is there any resize is pending on CVC if
+    so check corresponding CV resize status if it is success take a lease and
+    check is there any capacity difference in CV and CVC if there is no change
+    then update CVC status condition as resize success and release the lease.
+    If capacity change is noticed after taking lease, repeat this step after
+    releasing lease(left as is reconciliation will do rework).
+
+    Example status: CVC status when resize is success
+```yaml
+      Status:
+        Conditions:
+          - Type: Resizing
+            Status: Success
+            LastProbeTime: date
+            LastTransitionTime: (update while updating resize condition status)
+            Reason: Capacity changed
+            Message: Resize Success
+```
+   Note: Please refer CVC under Custom Resources used to resize cStor volume section
+         for entier CVC.
+
+- Cstor-volume-mgmt container as a target side car which is already having a
+  volume-controller watching on CV object will get update event (if missed at
+  the time of sync event) and volume-controller will process the request with following steps
+  - Volume-controller will process the CV update event in below steps
+    - Volume-controller will executes `istgtcontrol resize <size_unit>`
+      command(rpc call) if there is any resize status is pending on that
+      volume(based on resize status condition).
+    - If above rpc call success volume-controller will updates istgt.conf file
+      and update CVR capacity to latest size(with CV capacity). If volume
+      controller succeed in updating cvr and istgt.conf file then update
+      resize status as success in CV.
+    - If it is a failure response from rpc then generate kubernetes events and
+      do nothing just return(reconciliation will take care of doing the above
+      process again).
+- When cstor-istgt get a resize request it will trigger a resize request to zfs
+  on successful response cstor-istgt will return a success response to
+  cstor-volume-mgmt.
+  - cstor-istgt will disconnect the replicas from which we have not received the
+    response and then the successful response will send to the sidecar.
+    cstor-istgt will send updated size to replicas at the time of management
+    connection with the target.
+  Note: Processing `istgtcontrol resize` is a blocking call.
+
+- zfs will receives the resize request, it will resize the corresponding volume
+  and sent back the response to cstor-istgt(aka target).
+  Note:
+    In ephemeral case(temporary storage) when resize is success at the time of
+    node unavailable cases and when the node up need to take care of rebuilding
+    snapshots of different sizes.
+
+## High Level cStor Volume Resize Workflow
```

Comment:
See if the diagram in the link can be converted to ascii drawing and inserted into this section.

Reply:
Updated with the relative image link(It will load on the page itself). PTAL on https://github.com/mittachaitu/openebs/blob/resize_oep/contribute/design/1.x/csi/20190107-cStor-volume-resize.md#high-level-cstor-volume-resize-workflow